### PR TITLE
feat: Add analytics to amcl

### PR DESCRIPTION
**Pull request by grieneis**

### CODE REVIEW

File: move_base_msgs/msg/flatCov.msg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+float64 xx
+float64 xy
+float64 yy
+float64 tt
```
Comment (grieneis) [inline move_base_msgs/msg/flatCov.msg]: Add a new line

File: move_base_msgs/msg/cluster.msg [hunk 2]
```diff
@@ -0,0 +1,6 @@
+geometry_msgs/Pose mean
+move_base_msgs/flatCov cov
+float64 weight
+int32 num_samples
+
```
Comment (grieneis) [inline move_base_msgs/msg/cluster.msg]: too many new lines

File: amcl/src/amcl_node.cpp [hunk 3]
```diff
@@ -431,6 +437,9 @@ AmclNode::AmclNode() :
   tf_ = new TransformListenerWrapper();
 
   pose_pub_ = nh_.advertise<geometry_msgs::PoseWithCovarianceStamped>("amcl_pose", 2, true);
+  analytics_pub_ = nh_.advertise<move_base_msgs::amcl_analytics>("amcl_analytics",2,true);
```
Comment (grieneis) [inline amcl/src/amcl_node.cpp]: Generally prefer a space after a ,

File: amcl/src/amcl_node.cpp [hunk 4]
```diff
@@ -1273,17 +1282,22 @@ AmclNode::laserReceived(const sensor_msgs::LaserScanConstPtr& laser_scan)
...
+
     double max_weight = 0.0;
     int max_weight_hyp = -1;
     std::vector<amcl_hyp_t> hyps;
     hyps.resize(pf_->sets[pf_->current_set].cluster_count);
+    int total_num_samples = 0;
     for(int hyp_count = 0;
         hyp_count < pf_->sets[pf_->current_set].cluster_count; hyp_count++)
     {
+      move_base_msgs::cluster tC; //cluster to push back to data_msg.clusters
```
Comment (grieneis) [inline amcl/src/amcl_node.cpp]: Why tC for variable name?  camelCase doesn't match the surrounding style.  It's better to keep your code style similar to the code you are modifying

File: move_base_msgs/CMakeLists.txt [hunk 5]
```diff
@@ -27,12 +27,17 @@ add_message_files(
...
+        amcl_data.msg
+        cluster.msg
+        flatCov.msg
 )
 
 generate_messages(
     DEPENDENCIES
         actionlib_msgs
         geometry_msgs
+        move_base_msgs
```
Comment (grieneis) [inline move_base_msgs/CMakeLists.txt]: Was this necessary for compilation?  A dependency on itself?

**Review by grieneis — COMMENTED** [review]
Generally looks good.  A couple style comments.

File: move_base_msgs/CMakeLists.txt (hunk 5 above)
Comment (dnikitop) [reply move_base_msgs/CMakeLists.txt]: Had an early error with dependencies and wanted to see if messages need it for other messages in the same package. Do they?

**Review by ccpowers — APPROVED** [review]

**Pull request merged**